src/medVtkInria/medClutEditorToolBox/medClutEditorToolBox.cpp

```diff
@@ -370,71 +325,111 @@ void medClutEditorToolBox::invertLUT(bool value)
+            {
+                d->scene->table()->deleteAllVertices();
                 d->scene->removeItem(d->scene->table());
                 d->scene->addItem(table);
                 d->scene->update();
             }
         }
     }
-
 }
 
 void medClutEditorToolBox::onSaveTableAction(void)
 {
-
-    if ( medClutEditorTable * table = d->scene->table() ) {
+    if ( medClutEditorTable * table = d->scene->table() )
+    {
         medSaveLUTDialog dialog (table->title(),this);
+
         // Show it and wait for Ok or Cancel
-        if( dialog.exec() == QDialog::Accepted ) {
-            // We got Ok
+        if( dialog.exec() == QDialog::Accepted )
+        {
+            updateSavedTables();
+
+            // Copy the current table
             medClutEditorTable * tableToSave;
-            if (table->title().compare(dialog.textValue())) {
-                //copy the table
-                tableToSave = new medClutEditorTable( * table );
-                tableToSave->setTitle(dialog.textValue());
-            }
-            else {
-                tableToSave = table;
+            tableToSave = table;
+            tableToSave->setTitle(dialog.textValue());
+
+            // Check if table name already saved
+            int indexIfTableAlreadySaved = -1;
+            int currentIndex = 0;
+            foreach (medClutEditorTable* savedtable, *d->tables )
+            {
+                if (tableToSave->title() == savedtable->title())
+                {
+                    // The file needs to be updated, not append
+                    indexIfTableAlreadySaved = currentIndex;
+                    break;
+                }
+                currentIndex++;
             }
 
             QList<medClutEditorTable*>  l = *d->tables;
-            l << tableToSave;
+            if (indexIfTableAlreadySaved != -1)
+            {
+                l[indexIfTableAlreadySaved] = tableToSave; // Update
+            }
+            else
+            {
+                l << tableToSave; // Append
+            }
+
             medLUTToXMLWriter writer (l);
             //create file object in an existing storage place
             medStorage::mkpath(medStorage::dataLocation());
             QString fileName = medStorage::dataLocation();
             fileName.append("/LUTs.xml");
             QFile * file = new QFile();
             file->setFileName(fileName);
-            if (file->open(QIODevice::WriteOnly)) {
+
+            if (file->open(QIODevice::WriteOnly))
+            {
                 writer.writeFile(file);
                 file->close();
             }
-            else {
+            else
+            {
                 qDebug() << "can't open file " << fileName;
             }
+
             delete file;
+
+            updateSavedTables();
         }
     }
 }
 
-// void medClutEditorToolBox::onDeleteTableAction(void)
-// {
-
-// }
+void medClutEditorToolBox::updateSavedTables()
+{
+    d->tables = new QList<medClutEditorTable *>();
```

Comment:
There are some memory issues here coming from the previous code that was copied:

The `d->tables` member should not be a pointer but a direct QList (same for the tables member  in `medXMLToLUTReader`). Making it a pointer requires deleting it, which isn't done in the destructor or in the line above, causing memory leaks.

Also, `d->scene` is not deleted, and should not be a pointer.

Reply:
In fact, the QList shouldn't even contain pointers because they're not being freed either. It should be `QList<medCLutEditorTable>`.
(I know this is all pre-existing code but it seems to be full of bad memory handling)

Reply:
I would rather call it "Colormap Editor" because we don't edit the histogram, only how it is rendered in the view through colors and transparency.